services/src/test/java/io/grpc/protobuf/service/ProtoReflectionServiceTest.java

```diff
@@ -0,0 +1,312 @@
+                    .build())
+            .build();
+
+    StreamRecorder<ServerReflectionResponse> responseObserver = StreamRecorder.create();
+    StreamObserver<ServerReflectionRequest> requestObserver =
+        reflectionService.serverReflectionInfo(responseObserver);
+    requestObserver.onNext(request);
+    requestObserver.onCompleted();
+    assertEquals(goldenResponse, responseObserver.firstValue().get());
+  }
+
+  @Test
+  public void fileContainingSymbol() throws Exception {
+    final ServerReflectionRequest request =
+        ServerReflectionRequest.newBuilder()
+            .setHost(TEST_HOST)
+            .setFileContainingSymbol("grpc.reflection.testing.ReflectableService.Method")
+            .build();
+
+    final List<ByteString> goldenResponse =
+        Arrays.asList(
+            ReflectionTestProto.getDescriptor().toProto().toByteString(),
+            ReflectionTestDepthTwoProto.getDescriptor().toProto().toByteString(),
+            ReflectionTestDepthTwoAlternateProto.getDescriptor().toProto().toByteString(),
+            ReflectionTestDepthThreeProto.getDescriptor().toProto().toByteString()
+        );
+
+    StreamRecorder<ServerReflectionResponse> responseObserver = StreamRecorder.create();
+    StreamObserver<ServerReflectionRequest> requestObserver =
+        reflectionService.serverReflectionInfo(responseObserver);
+    requestObserver.onNext(request);
+    requestObserver.onCompleted();
+
+    final List<ByteString> response =
+        responseObserver.firstValue().get()
+            .getFileDescriptorResponse().getFileDescriptorProtoList();
+    assertEquals(goldenResponse.size(), response.size());
+    assertEquals(new HashSet<ByteString>(goldenResponse), new HashSet<ByteString>(response));
+  }
+
+  @Test
+  public void fileContainingNestedSymbol() throws Exception {
+    final ServerReflectionRequest request =
+        ServerReflectionRequest.newBuilder()
+            .setHost(TEST_HOST)
+            .setFileContainingSymbol("grpc.reflection.testing.NestedTypeOuter.Middle.Inner")
+            .build();
+
+    final ServerReflectionResponse goldenResponse =
+        ServerReflectionResponse.newBuilder()
+            .setValidHost(TEST_HOST)
+            .setOriginalRequest(request)
+            .setFileDescriptorResponse(
+                FileDescriptorResponse.newBuilder()
+                    .addFileDescriptorProto(
+                        ReflectionTestDepthThreeProto.getDescriptor().toProto().toByteString())
+                    .build())
+            .build();
+
+    StreamRecorder<ServerReflectionResponse> responseObserver = StreamRecorder.create();
+    StreamObserver<ServerReflectionRequest> requestObserver =
+        reflectionService.serverReflectionInfo(responseObserver);
+    requestObserver.onNext(request);
+    requestObserver.onCompleted();
+    assertEquals(goldenResponse, responseObserver.firstValue().get());
```

Comment:
Make sure to split tests into 3 distinct sections: Setup, test, verify.  Each section should be separated by a newline to make it visually easier to parse.

(It also has a nice side effect that if it is hard to fit your test into this structure, it means the test is testing too much.  You're code is fine here though)